Assets/HoloToolkit/Input/Scripts/Utilities/Interactions/TwoHandRotateLogic.cs

```diff
@@ -0,0 +1,153 @@
+        /// <returns>A Vector3 describing the direction of the line connecting the inputSources</returns>
+        private Vector3 GetHandlebarDirection(Dictionary<uint, Vector3> handsPressedMap, Transform manipulationRoot)
+        {
+            Assert.IsTrue(handsPressedMap.Count > 1);
+            var handsEnumerator = handsPressedMap.Values.GetEnumerator();
+            handsEnumerator.MoveNext();
+            var hand1 = handsEnumerator.Current;
+            handsEnumerator.MoveNext();
+            var hand2 = handsEnumerator.Current;
+
+            // We project the handlebar direction into camera space because otherwise when we move our body the handlebard will move even 
+            // though, relative to our heads, the handlebar is not moving.
+            hand1 = CameraCache.Main.transform.InverseTransformPoint(hand1);
+            hand2 = CameraCache.Main.transform.InverseTransformPoint(hand2);
+
+            return hand2 - hand1;
+        }
+
+        /// <summary>
+        /// TwoHandRotateLogic Constructor
+        /// </summary>
+        /// <param name="constrainRotation">Enum describing to which axis the rotation is constrained</param>
+        public TwoHandRotateLogic(AxisConstraint constrainRotation)
+        {
+            rotationConstraint = constrainRotation;
+        }
+
+        /// <summary>
+        /// Initializes twohand system with controller/hand source info: 
+        /// the Dictionary collection already filled with controller/hand info,
+        /// and the Transform of the GameObject to be manipulated.
+        /// </summary>
+        /// <param name="handsPressedMap">Dictionary listing inputSourceStates</param>
+        /// <param name="manipulationRoot">Transform of gameObject to be two hand manipulated</param>
+        public void Setup(Dictionary<uint, Vector3> handsPressedMap, Transform manipulationRoot)
+        {
+            currentRotationConstraint = rotationConstraint;
+            previousHandlebarRotation = GetHandlebarDirection(handsPressedMap, manipulationRoot);
+        }
+
+        /// <summary>
+        /// Updates internal states based on current Controller/hand states.
+        /// </summary>
+        /// <param name="handsPressedMap">Dictionary listing inputSourceStates</param>
+        /// <param name="manipulationRoot">Transform of gameObject to be two hand manipulated</param>
+        /// <param name="currentRotation">New rotation to be applied</param>
+        /// <returns>Quaternion describing rotation based on position of inputSources - Controllers/Hands</returns>
+        public Quaternion Update(Dictionary<uint, Vector3> handsPressedMap, Transform manipulationRoot, Quaternion currentRotation)
+        {
+            var handlebarDirection = GetHandlebarDirection(handsPressedMap, manipulationRoot);
+            var handlebarDirectionProjected = ProjectHandlebarGivenConstraint(currentRotationConstraint, handlebarDirection,
+                manipulationRoot);
```

Comment:
let's either put this on one line or break up each parameter on their own lines. Just needs to be consistent.